src/fmt.rs

```diff
@@ -0,0 +1,26 @@
+use crate::{StrIndex, StrRange};
+use std::fmt;
+
+impl fmt::Debug for StrIndex {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self)
```

Comment:
Will this handle padding and alignment automatically? Perhaps it”s better just to call fmt::Display::fmt(raw)?

Reply:
Yes, any writes in `Debug::fmt` are already padded/aligned; you go through the `f.debug_*` functions for something more complicated than a primitive. This `write!` call is equivalent to manually calling `Display::fmt`, but it's slightly more idiomatic to go through the `write!` call. The `format_args!` machinery is specifically optimized to disappear completely in the just-forward-one-`fmt`-call case.

Reply:
I decided to go ahead and make the forwards-to-`Display` obvious by just using `fmt::Display::fmt`.